### PR TITLE
CORE-6703 Merkle doc updates

### PR DESCRIPTION
**Pull request by vlajos**

### CODE REVIEW

File: crypto/src/main/kotlin/net/corda/v5/crypto/merkle/IndexedMerkleLeaf.kt [hunk 1]
```diff
@@ -2,6 +2,16 @@ package net.corda.v5.crypto.merkle
 
 import net.corda.v5.base.annotations.CordaSerializable
 
+/**
+ * [IndexedMerkleLeaf]s are building blocks of [MerkleProof]s.
+ * They contain the required information about a particular leaf which is needed for the verification.
+ *
+ * @property index The leaf's index.
+ * @property nonce The leaf's nonce.
```
Comment (mnesbit) [inline crypto/src/main/kotlin/net/corda/v5/crypto/merkle/IndexedMerkleLeaf.kt]: perhaps say: nonce the leaf's optional nonce.

Comment (vlajos) [reply crypto/src/main/kotlin/net/corda/v5/crypto/merkle/IndexedMerkleLeaf.kt]: Thanks. I have updated it.

**Review by mnesbit — DISMISSED** [review]
OK

File: crypto/src/main/kotlin/net/corda/v5/crypto/merkle/MerkleProof.kt [hunk 2]
```diff
@@ -10,22 +10,16 @@ import net.corda.v5.crypto.SecureHash
  * existing [MerkleTree].
  * Construct a [MerkleProof] from its ([treeSize], [leaves], [hashes]) when you want to [verify] if the leaves
  * to be checked are part of a [MerkleTree] with the specific root.
+ *
+ * @property treeSize Number of leaves in the whole tree
```
Comment (nadinequinr3) [inline crypto/src/main/kotlin/net/corda/v5/crypto/merkle/MerkleProof.kt]:
```suggestion
 * @property treeSize The total number of leaves in the tree.
```

File: crypto/src/main/kotlin/net/corda/v5/crypto/merkle/MerkleProof.kt [hunk 3]
```diff
@@ -10,22 +10,16 @@ import net.corda.v5.crypto.SecureHash
...
+ * @property leaves Leaf items whose inclusion is proved by the proof.
```
Comment (nadinequinr3) [inline crypto/src/main/kotlin/net/corda/v5/crypto/merkle/MerkleProof.kt]:
```suggestion
 * @property leaves The leaf items whose inclusion is proved by the proof.
```

File: crypto/src/main/kotlin/net/corda/v5/crypto/merkle/MerkleTree.kt [hunk 4]
```diff
@@ -7,21 +7,15 @@ import net.corda.v5.crypto.SecureHash
  * structured dataset.
  * From [MerkleTree]s, we can create [MerkleProof]s which let us prove that some particular data is part of the whole
  * tree without revealing the remaining data.
+ *
+ * @property leaves The input data elements. Usually something deterministically serialized.
+ * @property digestProvider [MerkleTreeHashDigestProvider] used to construct the tree's node and leaf hashes.
```
Comment (nadinequinr3) [inline crypto/src/main/kotlin/net/corda/v5/crypto/merkle/MerkleTree.kt]:
```suggestion
 * @property digestProvider The [MerkleTreeHashDigestProvider] used to construct the tree's node and leaf hashes.
```

**Review by nadinequinr3 — CHANGES_REQUESTED** [review]

**Review by mnesbit — APPROVED** [review]
OK

**Review by nadinequinr3 — APPROVED** [review]

**Pull request merged**